src/planner/binder/statement/bind_copy.cpp

```diff
@@ -182,7 +182,12 @@ BoundStatement Binder::BindCopyFrom(CopyStatement &stmt) {
 	insert.table = stmt.info->table;
 	insert.schema = stmt.info->schema;
 	insert.catalog = stmt.info->catalog;
-	insert.columns = stmt.info->select_list;
+	for (auto &expr : stmt.info->select_list) {
+		D_ASSERT(expr->type == ExpressionType::COLUMN_REF);
```

Comment:
Doesn't `Cast` effectively do this?

Reply:
I had a look, Cast will check the ExpressionClass, not the ExpressionType
I don't know enough about the semantical differences between those, so I think I'll keep the assert for now